src/runner/general_settings.cpp

```diff
@@ -214,6 +219,13 @@ void apply_general_settings(const json::JsonObject& general_configs, bool save)
         settings_theme = general_configs.GetNamedString(L"theme");
     }
 
+    if (json::has(general_configs, L"show_tray_icon", json::JsonValueType::Boolean))
+    {
+        show_tray_icon = general_configs.GetNamedBoolean(L"show_tray_icon");
+        // Update tray icon visibility when setting is toggled
+        set_tray_icon_visible(!show_tray_icon);
```

Comment:
```cpp
        // Update tray icon visibility when setting is toggled
        set_tray_icon_visible(show_tray_icon);
```

No longer needs to be inverted with the correction of the state change in `tray_icon.cpp`